### ISSUE TITLE
v2.6.1   campaign email send error -> Parse error: syntax error, unexpected '[' in ...

### ISSUE THREAD

**Issue by raymondkent** [mautic/mautic#3527]
#What type of report is this:
#3481 i think is related?

| Q  | A
| ---| ---
| Bug report? |  X
| Feature request? | 
| Enhancement? | 

## Description:
I am getting the following error when running 2.6.1 campaigns:trigger

[mautic_path]/app/console mautic:campaigns:trigger -f --campaign-id=1

Parse error: syntax error, unexpected '[' in [mautic_install_path]/app/bundles/ReportBundle/Model/ReportModel.php on line 708

  [Symfony\Component\Debug\Exception\FatalErrorException]
  Parse Error: syntax error, unexpected '['

mautic:campaigns:trigger [-i|--campaign-id [CAMPAIGN-ID]] [--scheduled-only] [--negative-only] [-l|--batch-limit [BATCH-LIMIT]] [-m|--max-events [MAX-EVENTS]] [-f|--force] [-t|--timeout TIMEOUT] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-s|--shell] [--process-isolation] [-e|--env ENV] [--no-debug] [--] <command>

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.6.1
| PHP version | 5.6.29

### Steps to reproduce:
1.  setup hostagtor cron settings to email output of cron jobs
2.  manually execute cron job from githib php script
3. open email and see error but have no idea how to fix.
 
i.e  If it try and figure it out based n the query pattern, should TIMEOUT and ENV both be enclosed in "[" and "]" ?

mautic:campaigns:trigger [-i|--campaign-id [CAMPAIGN-ID]] [--scheduled-only] [--negative-only] [-l|--batch-limit [BATCH-LIMIT]] [-m|--max-events [MAX-EVENTS]] [-f|--force] [-t|--timeout [TIMEOUT]] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-s|--shell] [--process-isolation] [-e|--env [ENV]] [--no-debug] [--] <command>

if that is the case, then how does one fix it?
or am I totally off track?

### Log errors: 

_Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable._ 

[2017-02-25 17:47:35] mautic.CRITICAL: Uncaught PHP Exception InvalidArgumentException: "Malformed UTF-8 characters, possibly incorrectly encoded" at [mautic_install_path]/vendor/symfony/http-foundation/JsonResponse.php line 164 {"exception":"[object] (InvalidArgumentException(code: 0): Malformed UTF-8 characters, possibly incorrectly encoded at [mautic_install_path]/vendor/symfony/http-foundation/JsonResponse.php:164)"} []
[2017-02-25 17:47:41] mautic.CRITICAL: Uncaught PHP Exception InvalidArgumentException: "Malformed UTF-8 characters, possibly incorrectly encoded" at [mautic_install_path]/vendor/symfony/http-foundation/JsonResponse.php line 164 {"exception":"[object] (InvalidArgumentException(code: 0): Malformed UTF-8 characters, possibly incorrectly encoded at [mautic_install_path]/vendor/symfony/http-foundation/JsonResponse.php:164)"} []
[2017-02-25 17:50:30] mautic.CRITICAL: Uncaught PHP Exception InvalidArgumentException: "Malformed UTF-8 characters, possibly incorrectly encoded" at [mautic_install_path]/vendor/symfony/http-foundation/JsonResponse.php line 164 {"exception":"[object] (InvalidArgumentException(code: 0): Malformed UTF-8 characters, possibly incorrectly encoded at [mautic_install_path]/vendor/symfony/http-foundation/JsonResponse.php:164)"} []
[2017-02-25 17:50:36] mautic.CRITICAL: Uncaught PHP Exception InvalidArgumentException: "Malformed UTF-8 characters, possibly incorrectly encoded" at [mautic_install_path]/vendor/symfony/http-foundation/JsonResponse.php line 164 {"exception":"[object] (InvalidArgumentException(code: 0): Malformed UTF-8 characters, possibly incorrectly encoded at [mautic_install_path]/vendor/symfony/http-foundation/JsonResponse.php:164)"} []

**Comment by escopecz** [comment]
See similar reports. For example https://github.com/mautic/mautic/issues/2198#issuecomment-239814170

**Comment by raymondkent** [comment]
Thanks John @escopecz 

I just checked and you are correct. The settings for the CLI PHP were 5.5 and worked fine for 2.5.1 version.

I changed the CLI to 5.6 and is now working again.

I don't recall seeing any advice when doing the upgrade that the CLI PHP needed to be 5.6 minimum. 

Is there anyway that when performing the upgrade that a check of the CLI PHP version can be made and if it doesn't meet the minimum then a warning occurs?

As mentioned the 2.5.1 worked with the CLI 5.5 , 

Thanks again for your feedback and assistance, I think I am back up and running.

**Comment by escopecz** [comment]
The requirement for PHP 5.6.19+ is there since Mautic 2.0.0. We follow semantic versioning, so supported PHP version, which is considered a backwards compatibility change won't change until 3.0.0 or more. See https://www.mautic.org/download/requirements/ for more details.

**Comment by raymondkent** [comment]
Thanks John.

I had 'most' of it correct, but it seems that when I upgraded my systems PHP from 5.5 to 5.6  when I moved to Mautic 2.x I did not change the cron job path from "55/bin" to "56/bin"- and everything still worked even in 2.5.1. 

Which is why this was all a bit confusing and frustrating when I upgraded to 2.6.1 - due to my own issues and not Mautics!

I really do appreciate your assistance in helping me get the system back up and running.

I think Mautic is awesome. Keep up the good work.

Sincerely

Ray
